python/pyiceberg/catalog/glue.py

```diff
@@ -0,0 +1,491 @@
+        )
+        io = load_file_io(properties=self.properties, location=metadata_location)
+        _write_metadata(metadata, io, metadata_location)
+
+        self._create_glue_table(
+            identifier=identifier, table_input=_construct_table_input(table_name, metadata_location, properties)
+        )
+        try:
+            loaded_table = self.load_table(identifier=(database_name, table_name))
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table {database_name}.{table_name} fail to be created") from e
+
+        return loaded_table
+
+    def load_table(self, identifier: Union[str, Identifier]) -> Table:
+        """Loads the table's metadata and returns the table instance.
+        You can also use this method to check for table existence using 'try catalog.table() except TableNotFoundError'
+        Note: This method doesn't scan data stored in the table.
+        Args:
+            identifier: Table identifier.
+        Returns:
+            Table: the table instance with its metadata
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            load_table_response = self.glue.get_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {database_name}.{table_name}") from e
+
+        return self._convert_glue_to_iceberg(load_table_response.get(PROP_GLUE_TABLE, {}))
+
+    def drop_table(self, identifier: Union[str, Identifier]) -> None:
+        """Drop a table.
+
+        Args:
+            identifier: Table identifier.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist, or the identifier is invalid
+        """
+        database_name, table_name = self.identifier_to_tuple(identifier)
+        try:
+            self.glue.delete_table(DatabaseName=database_name, Name=table_name)
+        except self.glue.exceptions.EntityNotFoundException as e:
+            raise NoSuchTableError(f"Table does not exists: {database_name}.{table_name}") from e
+
+    def purge_table(self, identifier: Union[str, Identifier]) -> None:
+        """Drop a table and purge all data and metadata files.
+
+        Args:
+            identifier (str | Identifier): Table identifier.
+
+        Raises:
+            NoSuchTableError: If a table with the name does not exist
+        """
+        table = self.load_table(identifier)
+        self.drop_table(identifier)
+        io = load_file_io(self.properties, table.metadata_location)
+        metadata = table.metadata
+        manifest_lists_to_delete = set()
+        manifests_to_delete = []
+        for snapshot in metadata.snapshots:
```

Comment:
I think we should move the logic of deleting the files into a function (`purge_files(table: Table)`?) in `__init__.py`. We can re-use this for the Hive catalog as well.

Reply:
Thank you for your suggestions. I moved the two helper functions, `delete_data_files` and `delete_files` to `__init__.py`. 
Also, I found that in [java impl](https://github.com/apache/iceberg/blob/a0a353d06d0095bbbf1bc46dab4c4122397bfe03/core/src/main/java/org/apache/iceberg/CatalogUtil.java#L188), we only log warning messages when encountering deletion failure. I wonder if it is ok that we do similar here? I used the `logger` defined in `catalog/__init__.py` to log the warning message.

Reply:
Yes, that looks good to me 👍🏻 We could even throw an exception.